### PR TITLE
[Triton/Gluon] [Config] Tune gfx1201 unified attention 2d for D=512 for full attention and D=256 for Sliding Window (Gemma4 shapes) and fix D=1024 launches

### PR DESCRIPTION
**Pull request by GodRishUniverse**
## Motivation

Improve unified-attention performance on Radeon AI PRO R9700 (gfx1201), especially in full attention. The workloads match Gemma-4-31B's full-attention head size of 512 and sliding-attention head size of 256.

## Technical Details

Only the gfx1201 unified-attention JSON changes. Add head-dimension and sliding-window selectors to choose workload-specific `attn_2d` settings.  **D** is head size (features per head), **Q** is maximum query length in the batch (tokens per sequence in this attention call), and **K** is KV sequence length. Query lengths are benchmark inputs, not fixed model properties.

For D>=1024, Q>=256, reduce `BLOCK_M` from 64 to 32, keeping tile=32. This fixes the observed D=1024 launch failure: 128 KiB shared memory requested against a 64 KiB per-workgroup budget on RDNA. 

Related PRs:

- **Merged:** [#4044](https://github.com/ROCm/aiter/pull/4044) documents Gemma-4 tuning on gfx950; [#4868](https://github.com/ROCm/aiter/pull/4868) adds RDNA **3D** LDS guards. This change targets gfx1201 **2D**.
- **Open tuning:** [#5601](https://github.com/ROCm/aiter/pull/5601) targets gfx1151; [#5650](https://github.com/ROCm/aiter/pull/5650) targets gfx942 and supersedes closed [#5063](https://github.com/ROCm/aiter/pull/5063); [#5432](https://github.com/ROCm/aiter/pull/5432) targets gfx950.
- **Open related fix:** [#5035](https://github.com/ROCm/aiter/pull/5035) is a draft gfx1201 **3D** LDS-budget fix.

## Test Plan

- Validate JSON and run `git diff --check`.
- Run the official parameterized correctness subset:
  `python -m pytest -q op_tests/triton_tests/attention/test_unified_attention.py -k '2048'`.
- Run `bench_unified_attention.py -test` for 20 targeted/boundary cases: Q=255/256/257 and 1023/1024/1025, sliding guards, and D=1024 full/sliding at Q=256/1024. Verify selected entries and reproduce the original launch failure.
- Compare original/current JSON using the same source and normal selection: five trials after prewarming, alternating run order, plus a sliding repeat. Baseline: `52ffe895e64acc993a14b6af04a588fc5a448d56`.

## Test Result

The parameterized correctness run completed with **no failures**:

- **512 passed:** executed tests matched the reference output within tolerance.
- **1536 skipped:** unavailable Gluon or shuffled-cache paths on gfx1201.
- **2144 deselected:** excluded by `-k '2048'`, which selects the smaller
  2048-block KV-cache cases that fit this GPU.

Separately, **all 20 targeted/boundary cases passed**. These checked the workloads being tuned, query lengths around configuration switches, sliding guards, and D=1024 full/sliding attention. The D=1024 cases now launch and match the reference; the original configuration's shared-memory failure was also reproduced. JSON validation and `git diff --check` passed.

Performance was measured on R9700 using **BF16 (bfloat16)**, comparing the original and tuned JSON with the same kernel code. Each value below is the **median of five runs per configuration**, after prewarming and with alternating run order. Speedup is baseline latency divided by tuned latency; lower latency is better. Every benchmark run also passed its reference correctness check.

Both workloads use 32 query heads. Full attention uses 4 KV heads and 64-token cache pages; sliding uses 16 KV heads, 32-token cache pages, and a 1024-token window. Cache page size is separate from the kernel's `BLOCK_M` setting.

| Workload | Baseline ms | Tuned ms | Speedup |
|---|---:|---:|---:|
| Full, D=512, Q=256, K=1024 | 1.539887 | 0.615243 | 2.50x |
| Full, D=512, Q=1024, K=8192 | 52.528322 | 18.196498 | 2.89x |
| Sliding, D=256, Q=1024, K=8192 | 0.764054 | 0.637692 | 1.20x |
| Sliding, same workload repeated | 0.723667 | 0.636494 | 1.14x |

Full attention was faster in every paired trial. Sliding was measured in two five-trial batches: the tuned config won 9 of 10 comparisons, but one was 1.67% slower. Its median gain was **1.14–1.20x**, with more variability than full attention. These results measure individual kernels, not whole-model throughput. D=1024 is reported as a launch fix because its baseline cannot run.

The unrestricted suite exceeded this 32 GiB GPU's memory while allocating test inputs, so this is a subset pass rather than a full-suite pass. Extra boundary checks use BF16. Note: The selector's `bf16` tag also includes FP16.

## Submission Checklist

- [X] Look over the contributing guidelines at https://github.com/ROCm/TheRock/blob/main/GOVERNANCE.md#pull-requests.

### CODE REVIEW

**Comment by GodRishUniverse** [comment]
Pinging @amd-xavierwang

File: aiter/ops/triton/configs/gfx1201/triton/attention/unified_attention/DEFAULT.json [hunk 1]
```diff
@@ -14,6 +16,54 @@
     ]
   },
   "attn_2d": {
+    "D_GEQ_1024.Q_GEQ_256.DT_bf16_bf16": {
```
[the remark this answers is not in the document]
Comment (GodRishUniverse) [reply aiter/ops/triton/configs/gfx1201/triton/attention/unified_attention/DEFAULT.json]: Should be fixed now

[the remark this answers is not in the document]
Comment (GodRishUniverse) [reply aiter/ops/triton/configs/gfx1201/triton/attention/unified_attention/DEFAULT.json]: Should be fixed now
